app/server/lib/DocStorage.ts

```diff
@@ -1568,6 +1570,30 @@ export class DocStorage implements ISQLiteDB, OnDemandStorage {
     return this._markAsChanged(db.requestVacuum());
   }
 
+  /**
+   * Run a VACUUM on the document and mark it as changed only
+   * if the saved space is above the ratio defined in SHRINK_RATIO_FOR_PUSH.
+   * Therefore if the doc storage is a remote one (like S3), it would be pushed
+   * only when we meet this condition, otherwise it is assumed not to be necessary.
+   */
+  public async vacuum(): Promise<void> {
+    const db = this._getDB();
+    const initSize = await this.storageManager.getFsFileSize(this.docName);
+    log.info(`Start Vacuum of doc ${this.docName}`);
```

Comment:
rawInfo is much more useful for "real" use, as opposed to watching the logs of a single document during development.

Reply:
Done in https://github.com/gristlabs/grist-core/pull/1624/commits/cc81a1d5612756bb4f303bb9e775d56a03116cd9